plugins/inputs/filestat/filestat.go

```diff
@@ -35,11 +35,18 @@ type FileStat struct {
 
 	// maps full file paths to globmatch obj
 	globs map[string]*globpath.GlobPath
+
+	// files that were missing - we only log the first time it's not found.
+	missingFiles map[string]bool
+	// files that had an error in Stat - we only log the first error.
+	filesWithErrors map[string]bool
```

Comment:
fyi, struct{} uses no memory.
```suggestion
	// files that were missing - we only log the first time it's not found.
	missingFiles map[string]struct{}
	// files that had an error in Stat - we only log the first error.
	filesWithErrors map[string]struct{}
```

Reply:
Agreed that a `struct{}` map would use less memory, but then again it'll use about as much memory as the number of files configured to be checked, whose config structure already use memory ... it comes out a wash, I feel like.